// adminjs/lib/frontend/interfaces/action/build-action-click-handler.js
/* eslint-disable no-restricted-globals */
/* eslint-disable no-undef */
/* eslint-disable no-alert */

import { actionHasDisabledComponent } from './action-has-component.js';
import { actionHref } from './action-href.js';
import { buildActionCallApiTrigger } from './build-action-api-call-trigger.js';
export const buildActionClickHandler = options => {
  const {
    action,
    params,
    actionResponseHandler,
    navigate,
    modalFunctions
  } = options;
  const {
    openModal
  } = modalFunctions;
  const handleActionClick = event => {
    event.preventDefault();
    event.stopPropagation();
    const href = actionHref(action, params);
    const callApi = buildActionCallApiTrigger({
      params,
      action,
      actionResponseHandler
    });

    // Action has "component" option set to "false" explicitly in it's configuration
    if (actionHasDisabledComponent(action)) {
      if (action.guard) {
        const modalData = {
          modalProps: {
            variant: 'danger',
            label: 'confirm',
            title: action.guard
          },
          type: 'confirm',
          resourceId: params.resourceId,
          confirmAction: callApi
        };

        // If confirmation is required, action trigger should be handled in modal
        openModal(modalData);
        return;
      }

      // If no confirmation is required, call API
      callApi();
      return;
    }

    // Default behaviour - you're navigated to action URL and logic is performed on it's route
    if (href) {
      navigate(href);
    }
  };
  return handleActionClick;
};
