// adminjs/lib/frontend/components/app/action-header/styled-back-button.js
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
import React, { useMemo } from 'react';
import { Link as RouterLink } from 'react-router-dom';
import { ButtonCSS, Icon } from '@adminjs/design-system';
import { styled } from '@adminjs/design-system/styled-components';
import { useSelector } from 'react-redux';
import allowOverride from '../../../hoc/allow-override.js';
// eslint-disable-next-line @typescript-eslint/no-unused-vars
const StyledLink = styled(({
  rounded,
  to,
  ...rest
}) => /*#__PURE__*/React.createElement(RouterLink, _extends({
  to: to
}, rest)))`${ButtonCSS}`;
const StyledBackButton = props => {
  const {
    showInDrawer
  } = props;
  const {
    previousRoute
  } = useSelector(state => state.drawer);
  const {
    from = {}
  } = useSelector(state => state.router);
  const cssCloseIcon = showInDrawer ? 'ChevronRight' : 'ChevronLeft';
  const backLink = useMemo(() => {
    if (!showInDrawer) {
      return from?.pathname;
    }
    if (previousRoute?.pathname) {
      return `${previousRoute?.pathname}${previousRoute?.search}`;
    }
    return from?.pathname;
  }, [previousRoute, from]);
  return /*#__PURE__*/React.createElement(StyledLink, {
    size: "icon",
    to: backLink,
    rounded: true,
    mr: "lg",
    type: "button"
  }, /*#__PURE__*/React.createElement(Icon, {
    icon: cssCloseIcon
  }));
};
const OverridableStyledBackButton = allowOverride(StyledBackButton, 'StyledBackButton');
export { OverridableStyledBackButton as default, OverridableStyledBackButton as StyledBackButton, StyledBackButton as OriginalStyledBackButton };
